Gems/Atom/Feature/Common/Assets/Materials/Types/Silhouette_DepthState.lua

```diff
@@ -0,0 +1,56 @@
+--------------------------------------------------------------------------------------
+--
+-- Copyright (c) Contributors to the Open 3D Engine Project.
+-- For complete copyright and license terms please see the LICENSE at the root of this distribution.
+--
+-- SPDX-License-Identifier: Apache-2.0 OR MIT
+--
+--
+--
+----------------------------------------------------------------------------------------------------
+
+ 
+function GetMaterialPropertyDependencies()
+    return {
+        "silhouetteType"
+        }
+end
+
+SilhouetteType_AlwaysDraw = 0
+SilhouetteType_Visible = 1
+SilhouetteType_XRay = 2
+SilhouetteType_NeverDraw = 3
+
+ComparisonFunc_Never = 0
+ComparisonFunc_Less = 1
+ComparisonFunc_Equal = 2
+ComparisonFunc_LessEqual = 3
+ComparisonFunc_Greater = 4
+ComparisonFunc_NotEqual = 5
+ComparisonFunc_GreaterEqual = 6
+ComparisonFunc_Always = 7
+
+function Process(context)
+    local silhouetteType = context:GetMaterialPropertyValue_enum("silhouetteType")
+    local shaderItem = context:GetShaderByTag("SilhouetteGather")
+    if(silhouetteType == SilhouetteType_AlwaysDraw) then
+        -- Always draws in ignores depth check, but won't draw where silhouettes are blocked
+        shaderItem:GetRenderStatesOverride():SetDepthEnabled(false)
+        shaderItem:GetRenderStatesOverride():SetDepthComparisonFunc(ComparisonFunc_Never)
+        -- if you want to let the full silhouette draw even on top of meshes that block
+        -- silhouettes, you can disable the stencil check using SetStencilEnabled()
+        -- e.g. shaderItem:GetRenderStatesOverride():SetStencilEnabled(false)
```

Comment:
Do we want to keep this in?

Reply:
I think it's useful to show how to modify the stencil from Lua if you want to make it so "AlwaysDraws" ignores silhouette blockers too.